api/v1alpha1/repository_types.go

```diff
@@ -28,14 +51,38 @@ type RepositorySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Repository. Edit repository_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// URL chart repository address
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
+
+	// BasicAuthSecret if the chart repository requires basic auth authentication,
+	// set the username and password to secret, with the fields user and password respectively.
+	BasicAuthSecret string `json:"basicAuthSecret,omitempty"`
+
+	// SkipTLS will not validate the repository's certificate
+	SkipTLS bool `json:"skipTLS,omitempty"`
+
```

Comment:
inspire by helmfile, Helm charts in OCI format will have some differences and need to be specified in the spec.
```go
// OCI repostiory
OCI bool `json:"oci,omitempty"`
```

Reply:
@Abirdcfly Does OCI format have any differences when  doing deployment?

Reply:
> OCI format have any differences when doing deployment

Take a quick look at helmfile's code, OCI url prefix will be `oci://`, qualifiedChartName will be different, and login also uses a different function.

**This can also be added later**. Let's talk about it after we have actually used OCI's helm repo.